src/NLog/Targets/MailTarget.cs

```diff
@@ -449,6 +451,26 @@ internal void ConfigureMailClient(LogEventInfo lastEvent, ISmtpClient client)
             
         }
 
+        /// <summary>
+        /// Handle <paramref name="pickupDirectoryLocation"/> if it is a virtual directory. Set PickupDirectoryLocation of <paramref name="client"/>
+        /// </summary>
+        /// <param name="client">client to set properties on</param>
+        /// <param name="pickupDirectoryLocation"></param>
+        /// <returns></returns>
+        internal static string GetPickupDirectoryLocation(ISmtpClient client, string pickupDirectoryLocation)
+        {
+            if (!pickupDirectoryLocation.StartsWith("~/"))
+            {
+                return pickupDirectoryLocation;
+            }
+
+            // Support for Virtual Paths
+            var root = AppDomain.CurrentDomain.BaseDirectory;
+            var directory = client.PickupDirectoryLocation.Replace("~/", "").Replace('/', Path.DirectorySeparatorChar);
```

Comment:
This could be improved:

- "~/" should be a `const string` (e.g. virtualPathPrefix)
- We check before `pickupDirectoryLocation` and not `client.PickupDirectoryLocation`
- We check a StartsWith but we're doing a replace all. We should replace only the first ~/ (or just `.Substring (virtualPathPrefix.length)` )